docs/src/introduction.md

```diff
@@ -58,6 +59,32 @@ p4est_connectivity_destroy(connectivity)
 ```
 
 
+## [`PointerWrapper`s](@id pointer_wrappers)
+
+As we have seen in the [previous section](@ref translation_guidelines) many functions
+provided by [`p4est`](https://github.com/cburstedde/p4est) return pointers to `struct`s,
+which requires to [`unsafe_load`](https://docs.julialang.org/en/v1/base/c/#Base.unsafe_load)
+the pointer in order to access the underlying data.
+[P4est.jl](https://github.com/trixi-framework/P4est.jl) offers a convenient way to avoid
+using `unsafe_load` whenever it is necessary by using a [`PointerWrapper`](@ref).
+If you, e.g., have a pointer to a `p4est_connectivity` (i.e. an object of type `Ptr{p4est_connectivity}`)
+called `connectivity`, you can use `connectivity_pw = PointerWrapper(connectivity)` to obtain
+a wrapped version of the pointer, where the underlying data can be accessed simply by
+`connectivity_pw.num_trees[]` without the need to call `unsafe_load` manually. This works even for nested
+structures, where the data of a `struct` is, again, a `struct`. The example from above, but now
```

Comment:
I don't understand the
> where the data of a `struct` is, again, a `struct`

part. Maybe you can give an example for a nested access to make it clearer?

Reply:
I should probably say "where a named field of a `struct` is a pointer to a `struct`", e.g., as in the struct for a `p4est`: It contains a field that is a `Ptr{p4est_connectivity_t}`. So what I want to say is that with a `PointerWrapper` it is possible to do something like:
```
connectivity = p4est_connectivity_new_brick(2, 2, 0, 0)
p4est = p4est_new_ext(MPI.COMM_WORLD, connectivity, 0, 0, true, 0, C_NULL, C_NULL)
p4est_pw = PointerWrapper(p4est)
p4est_pw.connectivity.num_trees[]
```
and one doesn't need constructions like `unsafe_load(unsafe_load(p4est).connectivity).num_trees`. I will extend the example so it also contains this case.

Reply:
I made the statement a little bit more precise and included a nested access in the example in [93fa7ea](https://github.com/trixi-framework/P4est.jl/pull/67/commits/93fa7ea2a0f2f7b8093a4fbae6798e3360e62b09).